src/SearchBook.js

```diff
@@ -0,0 +1,90 @@
+import React from "react"
+import {Link} from "react-router-dom"
+import Book from "./Book"
+
+import * as BooksAPI from './BooksAPI'
+
+class SearchBook extends React.Component {
+  state = {
+    enteredText: '',
+    queriedBooks: [],
+    myBooks: []
+  }
+  componentDidMount() {
+    BooksAPI.getAll().then(myBooks => {
+      this.setState({ myBooks })
+    })
+  }
+  handleChange = (enteredText) => {
+    this.setState(() => {return {enteredText: enteredText}})
+    this.search_books(enteredText)
+  }
+  search_books = (serchText) => {
+    if(serchText) {
+      BooksAPI.search(serchText).then((books) => {
+        if(books && books.length) {
+          books = books.map(book => {
+            book.shelf = this.addShelf(book)
+            return book
+          })
+
+          this.setState(() => {
+            return {queriedBooks: books}
+          })
+        } else {
+          this.setState(() => {
+            return {queriedBooks: []}
+          })
+        }
+      })
+    } else {
+      this.setState({queriedBooks: [], enteredText: ''})
+    }
+  }
+  addShelf(book) {
```

Comment:
Please consider changing the name of this method to a description that matches what it's doing or checking. It doesn't seem to be adding a book to a shelf or shelf to a book.

Reply:
@rogergraves  actually this method add`s the book for 'search shelf'...  I will rename this one